internal/cache/backend.go

```diff
@@ -174,25 +194,8 @@ func (b *Backend) Load(ctx context.Context, h restic.Handle, length int, offset
 		debug.Log("error loading %v from cache: %v", h, err)
 	}
 
-	// partial file requested
-	if offset != 0 || length != 0 {
```

Comment:
I just noticed that the current behavior is equivalent to removing this if and merging it into the if in line 195:
`if _, ok := autoCacheFiles[h.Type]; (offset != 0 || length != 0) && !b.Cache.PerformReadahead(h) || !ok`

As the index and snapshot files are always read completely this simplifies to `!ok && !b.Cache.PerformReadahead(h)` similar to the `autoCacheTypes` change. It would be good to keep this logic simplification.